BlogTemplate/Models/BlogDataStore.cs

```diff
@@ -323,9 +326,58 @@ public void UpdatePost(Post newPost, Post oldPost)
             SavePost(newPost);
         }
 
-        public bool CheckSlugExists(string slug)
+        public void SaveFiles(List<IFormFile> files)
+        {           
+            foreach(var file in files)
+            {
+                if(file.Length > 0)
+                {                    
+                    using (Stream uploadedFileStream = file.OpenReadStream())
+                    {
+                        byte[] buffer = new byte[uploadedFileStream.Length];
+                        uploadedFileStream.Read(buffer, 0, buffer.Length);
+                        string name = CreateFileName(file.FileName);
+                        _fileSystem.WriteFile($"{UploadsFolder}\\{name}", buffer);
+                    }
+                }
+            }
+        }
+
+        public IEnumerable<string> GetFileNames()
+        {
+            IEnumerable<string> fileNames = _fileSystem.EnumerateFiles(UploadsFolder);
+            return fileNames;
+        }
+
+        private bool CheckFileNameExists(string fileName)
         {
-            return _fileSystem.FileExists($"{PostsFolder}\\{slug}.xml");

```

Comment:
Is this no longer used for post files?  If it is still used, we may need separate methods for posts vs. uploads vs. drafts.